common/types/testutil.go

```diff
@@ -148,9 +148,11 @@ func GenLayerProposal(layerID LayerID, txs []TransactionID) *Proposal {
 					},
 				},
 			},
-			TxIDs: txs,
 		},
 	}
+	if len(txs) > 0 {
+		p.InnerProposal.TxIDs = txs
```

Comment:
if you need to change it here, why not in miner/proposal_builder.go too?

Reply:
I had a hare test failing due to `TxIDs: ([]types.TransactionID) {}` coming from this func and decoded by scale codec into `TxIDs: ([]types.TransactionID) <nil>` as there is no way for now to distinguish nil and empty slices in the codec. @dshulyak do we need the same change in other places (where codec is probably not applied as long as tests pass)?

Reply:
in some of the systests, we don't have any transactions in proposals. (TestAddNodes for example). you can verify whether that works, as proposals will be gossiped with no transactions inside.

Reply:
do we add/get to/from SQL database there with sql/proposals add/get? If not then it should not be a problem. I'll run systests anyway.

Reply:
> do we add/get to/from SQL database there with sql/proposals add/get?

i don't understand this question

Reply:
@evt in systest like TestAddNodes, we simply don't send transactions to the network, so mempool is empty => proposals don't contain any transactions